### ISSUE TITLE
Test suite does not declare expected failures correctly

### ISSUE THREAD

**Issue by mhsiah** [n1analytics/anonlink#62]
Running the test suite currently raises a RuntimeWarning:
```
/usr/lib/python3.6/unittest/case.py:550: RuntimeWarning: TestResult has no addExpectedFailure method, reporting as passes RuntimeWarning)
```
This is probably because the suite uses the Python `@unittest.expectedFailure` decorator instead of the Nose `@raises` decorator in various places (reasoning from [here](https://stackoverflow.com/a/37431536)).

**Comment by hardbyte** [comment]
As [Nose has been in maintenance mode for the past several years](http://nose.readthedocs.io/en/latest/) I switched the tests on [travis-ci](https://travis-ci.org/n1analytics/anonlink/jobs/342604434#L617) to use py.test - I suggest we solve this issue using the preferred py.test way.

**Comment by mhsiah** [comment]
Resolved by #62.
